skill-manager/skill_manager/routers/skill.py

```diff
@@ -286,26 +286,40 @@ async def query_skill(
     predictions = QueryOutput.parse_obj(response.json())
     logger.debug(
         "predictions from skill: {predictions}".format(
-            predictions=str(predictions.json())[:100]
+            predictions=str(predictions.json())
         )
     )
 
-    # save prediction to mongodb
-    mongo_prediction = Prediction(
-        skill_id=skill.id,
-        skill_name=skill.name,
-        query=query,
-        user_id=user_id,
-        predictions=predictions.predictions,
+    mongo_predictions = []
+    step = (
+        query_request.task_kwargs["topk"] if "topk" in query_request.task_kwargs else 1
     )
-    _ = mongo_client.client.skill_manager.predictions.insert_one(
-        mongo_prediction.mongo()
-    ).inserted_id
-    logger.debug(
-        "prediction saved {mongo_prediction}".format(
-            mongo_prediction=str(mongo_prediction.json())[:100],
+
+    # save prediction to mongodb
+    assert len(predictions.predictions) == len(queries) * step
+    for idx, query in enumerate(queries):
```

Comment:
Was passiert hier, wenn `queries` keine Liste von queries ist, sondern nur eine einzelne query?
Also quasi wenn es nicht `List[str]` ist, sondern halt `str`.

Reply:
Übersehen. Habe folgendes hinzugefügt:
```    
if isinstance(queries, str):
        queries = [queries]
```

Reply:
Ok. Das heißt dann dass `queries` dann einfach nur ein String ist? 
Oder ist es doch schon eine Liste mit halt nur einem Eintrag? 🤔 Vorher wurde ja auch `predictions=predictions.predictions,` gemacht.

Reply:
Queries ist entweder ein String oder eine Liste aus Strings. 
```
class QueryRequest(BaseModel):
    """The model for a query request that the skill receives."""

    query: Union[str, List[str]] = Field(
        ..., description="The input to the model that is entered by the user"
    )
    skill_args: Dict[str, Any] = Field(
        {}, description="Optional values for specific parameters of the skill"
    )
    skill: Dict[str, Any] = Field(
        {}, description="Skill information. See Skill-Manager for details."
    )
    user_id: str = Field("")
    explain_kwargs: Optional[Dict] = Field(
        {}, description="Optional values for obtaining explainability outputs."
    )
    attack_kwargs: Optional[Dict] = Field(
        {}, description="Optional values for obtaining adversarial outputs."
    )
    model_kwargs: Optional[Dict] = Field(
        {}, description="Optional values for the model forward pass."
    )
    task_kwargs: Optional[Dict] = Field({}, description="Optional values for the task.")
    preprocessing_kwargs: Optional[Dict] = Field(
        {}, description="Optional values for preprocessing."
    )
```
Wenn queries bspw vom evaluator kommt, ist es immer eine Liste aus Strings, egal ob jetzt eine query oder mehrere im request sind. Wenn der request aber nicht vom evaluator kommt, kann es entweder ein String oder eine Liste aus Strings sein. Im Falle, dass query ein string ist, müssen wir es wie du darauf hingewiesen hast in eine Liste packen.

Reply:
Okay, wenn das so klappt (habs jetzt nicht getestet), dann alles bestens 👍🏻